giraffe/src/components/BandLayer.tsx

```diff
@@ -0,0 +1,104 @@
+import * as React from 'react'
+import {useMemo, useRef, FunctionComponent} from 'react'
+
+import {LayerProps, BandLayerSpec, BandLayerConfig} from '../types'
+import {BandHoverLayer} from './BandHoverLayer'
+import {simplifyLineData} from '../utils/lineData'
+import {useCanvas} from '../utils/useCanvas'
+import {drawBands} from '../utils/drawBands'
+import {useHoverPointIndices} from '../utils/useHoverPointIndices'
+import {FILL} from '../constants/columnKeys'
+import {getBandHoverIndices} from '../utils/getBandHoverIndices'
+
+export interface Props extends LayerProps {
+  spec: BandLayerSpec
+  config: BandLayerConfig
+}
+
+export const BandLayer: FunctionComponent<Props> = props => {
+  const {config, spec, width, height, xScale, yScale, hoverX, hoverY} = props
+
+  const simplifiedLineData = useMemo(
+    () => simplifyLineData(spec.lineData, xScale, yScale),
+    [spec.lineData, xScale, yScale]
+  )
+
+  const drawBandsOptions = {
+    bandFillColors: spec.bandFillColors,
+    fill: spec.columnGroupMaps.fill,
+    interpolation: config.interpolation,
+    lineData: simplifiedLineData,
+    lineWidth: config.lineWidth,
+    lineOpacity: config.lineOpacity,
+    shadeBelowOpacity: config.shadeBelowOpacity,
+  }
+
+  const canvasRef = useRef<HTMLCanvasElement>(null)
+
+  useCanvas(
+    canvasRef,
+    width,
+    height,
+    context => drawBands({context, ...drawBandsOptions}),
+    Object.values(drawBandsOptions)
+  )
+
+  let hoverDimension: 'x' | 'y' | 'xy'
+
+  if (
+    config.hoverDimension === 'auto' &&
+    Object.keys(spec.lineData).length > config.maxTooltipRows
+  ) {
+    hoverDimension = 'xy'
+  } else if (config.hoverDimension === 'auto') {
+    hoverDimension = 'x'
+  } else {
+    hoverDimension = config.hoverDimension
+  }
+
+  const groupColData = spec.table.getColumn(FILL, 'number')
+  const hoverRowIndices = useHoverPointIndices(
+    hoverDimension,
+    hoverX,
+    hoverY,
+    spec.table.getColumn(config.x, 'number'),
+    spec.table.getColumn(config.y, 'number'),
+    groupColData,
+    xScale,
+    yScale,
+    width,
+    height
+  )
+
+  const bandHoverIndices = getBandHoverIndices(
+    hoverRowIndices,
+    groupColData,
+    spec.bandIndexMap
+  )
+
+  const hasHoverData = hoverRowIndices && hoverRowIndices.length > 0
+
+  return (
+    <>
+      <canvas
+        className="giraffe-layer giraffe-layer-band-chart"
+        ref={canvasRef}
+        style={{
+          position: 'absolute',
+          opacity: hoverDimension === 'xy' && hasHoverData ? 0.4 : 1,
```

Comment:
I'm not sure exactly what the issue is in the code, but when I change the hover dimension to 'xy' and try to hover over the graph, the graph disappears. Will changing the hover dimension be an option in the UI or is this just a giraffe issue?
![ezgif-6-ea5b3ef913cd](https://user-images.githubusercontent.com/34894628/89931069-b4725780-dbd9-11ea-8493-70ad5d77fa70.gif)

Reply:
Good catch. This is band chart specific. I need to update the hover points when the dimension is 'xy'

Reply:
Turns out, hovering for band charts is a LOT more complex. It wasn't just an issue with the 'xy' dimension. All dimensions had issues, including showing incorrect data points.

This is fixed now for all dimensions.